crates/services/executor/src/executor.rs

```diff
@@ -650,6 +655,48 @@ where
         Ok(data)
     }
 
+    fn process_da(
+        &self,
+        block_st_transaction: &mut D,
+        header: &PartialBlockHeader,
+    ) -> ExecutorResult<()> {
+        let block_height = *header.height();
+        let prev_block_height = block_height
+            .pred()
+            .ok_or(ExecutorError::ExecutingGenesisBlock)?;
+
+        let prev_block_header = block_st_transaction
+            .storage::<FuelBlocks>()
+            .get(&prev_block_height)?
+            .ok_or(ExecutorError::PreviousBlockIsNotFound)?;
+        let previous_da_height = prev_block_header.header().da_height;
```

Comment:
This made me smile. Get the current height -> get pred height -> get the da_block for pred height -> get da_block height. The pred height and the da_block height should be the same, but I guess you still want to do the `get` to make sure it exists.

I guess if there was an `exists(&prev_block_height)` it could be more explicit, but I don't care that much.

Reply:
The code here:
1. Get the current block height
2. Get the block height of the previous block(current - 1)
3. We get the da block height of the previous block.
4. We get the next unprocessed da block height starting from the previous block(prev_da_block_height + 1).
5. We process all DA blocks between `next_unprocessed_da_block_height`..=`current_da_block_height`.

We need to do something like this to handle cases:
1. The DA block height didn't change between fuel blocks.
2. The DA block height changed increased by X, where X > 0(it can be any value like 1, 10, or 1000000).

Reply:
OOooh. I misread one of the lines. This looks right.